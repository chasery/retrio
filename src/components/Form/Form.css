.Form {
  flex-grow: 1;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.0625rem #0000001c;
  border-radius: 0.125rem;
  min-width: calc(100% - 3rem);
  overflow: hidden;
}

.Form__header {
  padding: 1rem 1.5rem;
  background-color: #ffffff;
  color: #474747;
}

.Form__body {
  padding: 1rem 1.5rem;
}
.Form__body > p {
  margin-top: 0;
}

.Form__controls {
  display: flex;
  justify-content: flex-end;
  margin-top: 1.5rem;
}
.Form__controls > button + button {
  margin-left: 0.5rem;
}
.Form__button {
  display: block;
  padding: 0.5rem 1rem;
  color: #ffffff;
  border: none;
  border-radius: 0.125rem;
  text-decoration: none;
}
.Form__button:disabled {
  opacity: 0.5;
}
.Form__button:hover {
  box-shadow: inset 0 0.0625rem 0.0625rem #0000001c;
}
.Form__button.primary {
  background-color: #474747;
}
.Form__button.primary:hover {
  background-color: #ebebeb;
  color: #474747;
}
.Form__button.primary:disabled:hover {
  background-color: #474747;
  color: #ffffff;
}
.Form__button.secondary {
  background-color: transparent;
  color: #474747;
}
.Form__button.secondary:hover {
  background-color: #ebebeb;
}
.Form__button.secondary:disabled:hover {
  background-color: transparent;
  color: #474747;
}
.Form__button.action {
  background-color: #58bc82;
  color: #fffdff;
}
.Form__button.action:hover {
  background-color: #46af72;
}

.Form__footer {
  margin: 0;
  padding: 0.5rem 1.5rem 1.5rem 1.5rem;
}

.Form .Error {
  margin: 0.5rem 0 0;
}
.Form .Error.inline {
  margin: 0.5rem 0;
}
.Form .Error + .Form__controls {
  margin-top: 1rem;
}

@media all and (min-width: 30rem) {
  .Form {
    min-width: 24rem;
    max-width: 24rem;
  }
}
