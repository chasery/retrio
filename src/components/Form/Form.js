import React from "react";
import "./Form.css";

function Form(props) {
  return (
    <form id={props.id} className="Form" onSubmit={props.onSubmit}>
      {props.children}
    </form>
  );
}

export default Form;
