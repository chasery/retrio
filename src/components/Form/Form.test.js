import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Form from "./Form";

describe("Form component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<Form />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
