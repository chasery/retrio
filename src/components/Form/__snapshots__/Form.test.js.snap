// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders without error 1`] = `
<form
  className="Form"
/>
`;
